### PR TITLE
Pre commit setup

### PR DESCRIPTION
**Pull request by KyleQianliMa**
Created .pre-commit-config.yaml file. Ignored a few errors. It's better to correct these errors after I can install and run tests to prevent accidentally breaking something.

To install: (git checkout -b Pre-commit-Setup origin/Pre-commit-Setup)
1. set up conda environment "conda env create" using environment.yml file.
2. conda activate histogram-dev
3. in terminal, go into python/ locate setup.py. Run pip install -e. in this terminal

To test:
cd ~/histogram-dev/tests/histogram
run alltests.py either in VS code or terminal: python alltests.py. All tests should pass in tfile.

cd ~/historgram-dev/tests/histogram/ndarray
run alltests.py either in VS code or terminal: python alltests.py. Expect run 28 tests, 5 would fail.

Failing tests:
1.ERROR: testNumpyNdArray2StdVectorNdArray (converters_TestCase.converters_TestCase)
histogram.ndarray.converters: NumpyNdArray2StdVectorNdArray

2.ERROR: testStdVectorNdArray2NumpyNdArray (converters_TestCase.converters_TestCase)
histogram.ndarray.converters: StdVectorNdArray2NumpyNdArray

3.ERROR: test__div__ (NdArray_TestCase.createTestCase.<locals>.TC)
NdArray: operator 'a/b'

4.ERROR: test_dump (NdArray_TestCase.createTestCase.<locals>.TC)
NdArray: pickle.dump

5.ERROR: test_load (NdArray_TestCase.createTestCase.<locals>.TC)
NdArray: pickle.load

(Currently next(default), main, qa are the same as danse-inelastic/histogram:master)

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#default environment for installing python package
+#Note: Use this
```
Comment (mpatrou) [inline environment.yml:2]: the comments can be omitted.

Comment (KyleQianliMa) [reply environment.yml]: removed the comments and deleted the line with histogram

File: python/histogram/Histogram.py [hunk 2]
```diff
@@ -535,207 +576,199 @@ def transpose(self, *axis):
...
-
-    def axisFromName( self, name):
-        return self._axisCont.datasetFromName( name)
-        
+        try:
+            return names.index(name)
+        except:
+            raise UnknownAxis("%r. Axes: %s" % (name, self.axisNameList()))
 
-    def data( self):
```
Comment (mpatrou) [inline python/histogram/Histogram.py:717]: between lines 717-724 was there a renaming of the function?

File: python/histogram/_units.py [hunk 3]
```diff
@@ -1,24 +1,32 @@
 import numpy as N
 from pyre.units import *
```
Comment (mpatrou) [inline python/histogram/_units.py:2]: from pyre.units import * . thi can be omitted and import only the specific required functions

**Review by mpatrou — APPROVED** [review]
it looks good!!

**Pull request merged**